### PR TITLE
k8s.io/apiserver/storage: add 3 new unit tests for delete.

### PR DESCRIPTION
**Pull request by siyuanfoundation**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
More unit test coverage.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go [hunk 1]
```diff
@@ -370,6 +370,52 @@ func RunTestDeleteWithSuggestionAndConflict(ctx context.Context, t *testing.T, s
 	if err := store.Get(ctx, key, storage.GetOptions{}, &example.Pod{}); !storage.IsNotFound(err) {
 		t.Errorf("Unexpected error on reading object: %v", err)
 	}
+	updatedPod.ObjectMeta.ResourceVersion = "4"
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go:373]: Please don't set a constant here - instead set `out.ObjectMeta.ResourceVersion`

File: staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go [hunk 2]
```diff
@@ -370,6 +370,52 @@ func RunTestDeleteWithSuggestionAndConflict(ctx context.Context, t *testing.T, s
...
+		t.Errorf("Expect validateCount = %d, but got %d", 2, validateCount)
+	}
+	if updateCount != 1 {
+		t.Errorf("Expect updateCount = %d, but got %d", 1, updateCount)
+	}
+
+	if err := store.Get(ctx, key, storage.GetOptions{}, &example.Pod{}); !storage.IsNotFound(err) {
+		t.Errorf("Unexpected error on reading object: %v", err)
+	}
+	updatedPod.ObjectMeta.ResourceVersion = "4"
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go:417]: same here

**Review by wojtek-t — COMMENTED** [review]
Just two nits - other than that LGTM.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go (hunk 1 above)
Comment (siyuanfoundation) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go]: done.

File: staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go [hunk 3]
```diff
@@ -468,6 +570,35 @@ func RunTestPreconditionalDeleteWithSuggestion(ctx context.Context, t *testing.T
 	}
 }
 
+func RunTestPreconditionalDeleteWithOnlySuggestionPass(ctx context.Context, t *testing.T, store storage.Interface) {
```
Comment (logicalhan) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go:573]: What does `OnlySuggestionPass` mean?

Comment (siyuanfoundation) [reply staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go:578]: RunTestPreconditionalDeleteWithOnlySuggestionPass tests the case of delete with preconditions, when the suggested cachedExistingObject passes the preconditions while the current version does not pass the preconditions.

**Comment by wojtek-t** [comment]
Thanks for extending our test coverage!

/lgtm
/approve

**Pull request merged**